Sources/MongoMobile/MongoMobileLogger.swift

```diff
@@ -0,0 +1,99 @@
+/// If the user provided a logger when initializing the library, executes the 
+/// `onMessage` callback for that logger.
+internal func mongo_mobile_log_callback(userDataPtr: UnsafeMutableRawPointer?,
+                                        messagePtr: UnsafePointer<Int8>?,
+                                        componentPtr: UnsafePointer<Int8>?,
+                                        contextPtr: UnsafePointer<Int8>?,
+                                        severity: Int32) {
+    guard let logger = MongoMobile.logger else {
+        return
+    }
+
+    let message = String(from: messagePtr)
+    let component = String(from: componentPtr)
+    let context = String(from: contextPtr)
+    // swiftlint:disable:next force_unwrapping - server should always gives a valid raw value.
+    let severity = LogSeverity(rawValue: severity)!
+    logger.onMessage(message: message, component: component, context: context, severity: severity)
+}
+
+extension String {
+    /// Returns a string created from the given pointer, or an empty
+    /// string if the pointer is nil.
+    fileprivate init(from ptr: UnsafePointer<Int8>?) {
```

Comment:
Don't we have better [ways](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/MongoError.swift#L281-L290) of doing this?